src/int128_struct_impl.h

```diff
@@ -155,8 +155,14 @@ static SECP256K1_INLINE void secp256k1_i128_rshift(secp256k1_int128 *r, unsigned
    }
 }
 
+static SECP256K1_INLINE uint64_t secp256k1_i128_to_u64(const secp256k1_int128 *a) {
+   return a->lo;
+}
+
 static SECP256K1_INLINE int64_t secp256k1_i128_to_i64(const secp256k1_int128 *a) {
-   return (int64_t)a->lo;
+   /* Verify that a represents a 64 bit signed value by checking that the high bits are a sign extension of the low bits. */
+   VERIFY_CHECK(a->hi == -(uint64_t)(0x8000000000000000u == (0x8000000000000000u & a->lo)));
```

Comment:
```suggestion
   VERIFY_CHECK(a->hi == -(a->lo >> 63));
```
I find this easier to read (but it may just be me)

Reply:
Thank you.  I was fishing around for a good way to write this.